ext/standard/tests/streams/gh15937.phpt

```diff
@@ -0,0 +1,14 @@
+--TEST--
+GH-15937 (stream overflow on timeout setting)
+--FILE--
+<?php
+$config = [
+    'http' => [
+        'timeout'		=> PHP_INT_MAX,
+    ],
+];
+$ctx = stream_context_create($config);
+var_dump(fopen("http://example.com", "r", false, $ctx));
+?>
+--EXPECTF--
+resource(%d) of type (stream)
```

Comment:
It would be good if we can actually check the behavior here. I'm not sure what happens – is the timeout ignored? If there is no way to check that in userland, adding something to zend_tests might be possible, but may be overkill. Not sure.

Reply:
timeout < 0 (INFTIME) means blocking mode, see even for our [own emulation mode](https://github.com/php/php-src/blob/e2da65de2acae5eb17de4dd6a34bd1f8f5d8c007/main/network.c#L1223).